packages/aws_common/lib/src/js/common.dart

```diff
@@ -374,3 +374,14 @@ Object? dartify(Object? o) {
   }
   return o;
 }
+
+/// Convert a dart map to a JS object
+Object mapToJSObj(Map<Object, Object> a) {
```

Comment:
@dnys1 I tried using the _convertToJs but that didn't work so I created this method instead.

Reply:
What didn't work? This is basically the same. Also, it should be marked `@internal` if it's not private.

Reply:
Crashes code with exception: 

>   DataCloneError: Failed to execute 'add' on 'IDBObjectStore': _installSpecializedAsCheck(object) {
>         var t1, testRti = this,
>           asFn = A._generalAsCheckImplementati...<omitted>... } could not be cloned.
>   string_database_test.dart.browser_test.dart.js 7793:31   Object.callMethod
>   string_database_test.dart.browser_test.dart.js 22904:67  <fn>
> 

Reason: 

https://github.com/dart-lang/sdk/issues/34130

![image](https://user-images.githubusercontent.com/11478656/217264021-72c65287-6fa5-4597-9a84-ab5758e6c35a.png)

Reply:
It sounds like you were passing the `Map` without calling `jsify`. Your helper function here is exactly what `jsify` does: https://github.com/dart-lang/sdk/blob/15b77bb03c073fb0613873b7ff395addfe0bb03c/sdk/lib/_internal/js_shared/lib/js_util_patch.dart#L28

Please double check this is needed.